includes/discovery/sensors/voltage/cyberpower-ups.inc.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+echo 'Cyberpower UPS ';
+
+//Currently only supports single-phase devices
+
+$oid_array = [
+    [
+        'oidname'     => 'upsAdvanceOutputVoltage',
+        'type'        => 'cyberpower-ups',
+        'index'       => 0,
+        'descr'       => 'Output',
+        'divisor'     => 10,
+        'mib'         => '+CPS-MIB',
+    ],
+    [
+        'oidname'     => 'upsAdvanceInputLineVoltage',
+        'type'        => 'cyberpower-ups',
+        'index'       => 0,
+        'descr'       => 'Input',
+        'divisor'     => 10,
+        'mib'         => '+CPS-MIB',
+    ],
+];
+foreach ($oid_array as $item) {
+    $oids = snmp_get($device, $item['oidname'].'.'.$item['index'], '-OsqnU', $item['mib']);
+    d_echo($oids."\n");
+
+    $value = explode(" ", $oids);
+    $current = $value[1]/$item['divisor'];
+    $current_oid = $value[0];
+
+    if (!empty($oids)) {
+        d_echo($oids);
+        $oids = trim($oids);
+        if ($oids) {
+            echo $item['type'].' '.$item['mib'];
+        }
+        discover_sensor($valid['sensor'], 'voltage', $device, $current_oid, $current_oid, $item['type'], $item['descr'], $item['divisor'], 1, null, null, null, null, $current);
```

Comment:
This part of the discovery seems to be doable via the YAML file as far as I see it. Did you try this approach (and revert it for a reason ?)

Reply:
@PipoCanaja I just realized this is all happening because the UPS is not responding to snmpbulkwalk polls. Setting nobulk: true in cyberpower.yaml resolves the issue w/o need for further modifications. We should close out this PR. :)